### PR TITLE
Add `html` kwarg for loading article content

### PR DESCRIPTION
**Pull request by lwm**
My personal experience, #74, #161 and #37 bring me to believe this is necessary.

As it stands right now, I don't understand why you force the user to `download()`.

It's a confusing API (using `set_html`) for people who just want the parse functionality.

### CODE REVIEW

**Comment by lwm** [comment]
bump!

File: newspaper/article.py [hunk 1]
```diff
@@ -89,14 +89,20 @@ def __init__(self, url, title='', source_url='', config=None, **kwargs):
...
         self.article_html = ''
 
         # Flags warning users in-case they forget to download() or parse()
         # or if they call methods out of order
         self.is_parsed = False
+
         self.is_downloaded = False
+        if self.html:
+            # User has loaded html content, does not require download restriction
+            self.is_downloaded = True
```
Comment (codelucas) [inline newspaper/article.py]: Why did you add `self.is_downloaded = True`? set_html sets this to False as intended

**Comment by codelucas** [comment]
I added inline comments on why you set the is_downloaded flag. Is there a big need for this API change? I don't want to change the API often after this has been out so long unless there is a real need

File: newspaper/article.py (hunk 1 above)
Comment (lwm) [inline newspaper/article.py]: If the user passes some value for `self.html`, then we want to pass the `is_downloaded` check.

**Comment by lwm** [comment]
>  Is there a big need for this API change?

You had 3 issues and I also ran into the issue. More will come.

>  I don't want to change the API often after this has been out so long

This isn't a breaking change. It just allows people *not* using the download path.
